src/Image/Point.php

```diff
@@ -42,8 +42,8 @@ public function __construct($x, $y)
             throw new InvalidArgumentException(sprintf('A coordinate cannot be positioned outside of a bounding box (x: %s, y: %s given)', $x, $y));
         }
 
-        $this->x = $x;
-        $this->y = $y;
+        $this->x = ceil($x);
+        $this->y = ceil($y);
```

Comment:
I think the solution outlined in https://github.com/php-imagine/Imagine/pull/843#discussion_r1143490195 would be the correct fix here.